### ISSUE TITLE
What ports does the Keybase client need access to?

### ISSUE THREAD

**Issue by Tristor** [keybase/client#2832]
I'm getting an 
```
Error in transport: operation timed out
asynchronous t.run() failed with operation timed out
``` 
on a system which otherwise functions fine with Keybase when attempting to authenticate a new device.  It appears to succeed, but the new device can never login despite being shown in my device list.  This tells me that the key exchange failed.

I have extremely restrictive inbound and outbound firewall rules in `pf` on both devices, so am trying to determine what ports I need to open to allow Keybase to work properly as I assume this is the cause of the issue.

Thanks.

**Comment by maxtaco** [comment]
We don't listen on any incoming ports, we just make outgoing connections to `api.keybase.io:443` and `mdserver.kbfs.keybase.io:443` and `bserver.kbfs.keybase.io:443`. 

You might have uncovered a bug, but it's a bit hard to know without more data. Can you do `keybase log send`? 

CC: @patrickxb

**Comment by tduzan** [comment]
I just submitted the logs from my work PC:

```
Success! Your log ID is:

  c9e1956e1d86137c65ab5e1c
```

It does look like it was missing some logs it was looking for based on the warning messages so I'm not sure they'll be much of value.  Let me know if there's something else I can try here.  I'm using Little Snitch + Murus (pf) to filter both inbound and outbound traffic, and I've obviously got TCP 443 allowed outbound in pf and have whitelisted `keybase` accessing the domain `keybase.io` on port 443 in Little Snitch, so there shouldn't be any interference if that's all that is required.

**Comment by maxtaco** [comment]
The exchange protocol worked fine. Can you send logs from the device that can't log in?

**Comment by maxtaco** [comment]
BTW, here are your provisioned devices: https://keybase.io/tristor/devices

**Comment by maxtaco** [comment]
Looks like you successfully provisioned those devices and then revoked them.

**Comment by tduzan** [comment]
Yeah, I revoked them in response to the fact that after the provisioning appeared to complete (despite the timeout errors) that I was unable to login on the newly provisioned device with that device key... I get prompted to create it again.

Let me grab the other system and I will shortly provide logs from it as well.

**Comment by Tristor** [comment]
```
Success! Your log ID is:

  6a0cd0f058856dbef60e621c
```

Logs have been sent from this system (the one I'm trying to add).  Let me know if that pinpoints exactly what I'm doing wrong here or if there's a bug being triggered.

**Comment by Tristor** [comment]
The thing in particular which is interesting is when I do this process, from this system I see:
```
Enter a public name for this device: TrMBP15
✔ Success! You provisioned your device TrMBP15.

You are logged in as tristor
  - type `keybase help` for more info.
```
then

```
❯ keybase status
Username:      tristor
Logged in:     no
Session:       tristor [salt only]
    is valid:  no
    keys:      locked
    lksec:     loaded
```

**Comment by Tristor** [comment]
Trying to run `keybase login` again tries to go through the whole device provisioning process all over again...

**Comment by maxtaco** [comment]
What's in `/Users/tduzan/Library/Application Support/Keybase/config.json`?  That's all public information, there are no secrets in there.

cc: @patrickxb looks like a half-provisioned device like we saw a few weeks ago

**Comment by Tristor** [comment]
```
❯ cat /Users/tristor/Library/Application\ Support/Keybase/config.json
{
    "current_user": "tristor",
    "users": {
        "tristor": {
            "device": null,
            "id": "3bdaf8163f657d7fd8d259b0c4fef000",
            "name": "tristor",
            "salt": "86002a3bb0a7010f1cfd3d56c58f3097"
        }
    }
}
```

**Comment by maxtaco** [comment]
What if you change that `null` device ID to `5254f43598e045b0debff575ef9ad018`?

**Comment by maxtaco** [comment]
(and then try `keybase logout` followed by `keybase login`?)

**Comment by Tristor** [comment]
Same result, a `keybase login` triggers an attempt to provision this device again.

**Comment by maxtaco** [comment]
OK try these commands:
```
keybase config set --string users.tristor.device 5254f43598e045b0debff575ef9ad018
keybase config get users.tristor.device
keybase logout
keybase config get users.tristor.device
keybase login
keybase config get users.tristor.device
```

**Comment by Tristor** [comment]
That appears to have resolved the issue.  After prompting me via PinEntry for my Keybase password it now has a valid session.

```
❯ keybase status
Username:      tristor
Logged in:     yes

Device:
    name:      TrMBP15
    ID:        5254f43598e045b0debff575ef9ad018
    status:    active

Session:       tristor [salt only]
    is valid:  yes
    keys:      unlocked
    lksec:     loaded
```

Any thoughts as to how this happened and how I can prevent it in the future?  Also, is there anyway I can get one of the previously used device names back that I revoked before?  Unfortunately it looks like the device names must be unique and revoking them doesn't remove them.  Not the end of the world, but I would like something more descriptive.

**Comment by maxtaco** [comment]
We'll have to look into this brutal bug and comb through your logs.  In the mean time, glad it worked out.  You can't reuse device names unfortunately, it's a security precaution.

**Comment by Tristor** [comment]
That's understandable.  Thanks for your assistance @maxtaco

**Comment by maxtaco** [comment]
Thank you for all of your help and patience!
